lighthouse-core/scripts/chromium-web-tests/README.md

```diff
@@ -0,0 +1,57 @@
+# Chromium Web Tests
+
+This runs the Chromium webtests using the devtools integration tester.
+
+`webtests/` contains all of our Lighthouse webtests.
+
+## Run
+
+```sh
+yarn build-devtools
+node lighthouse-core/scripts/chromium-web-tests/test.sh
+
+# Reset the results.
+node lighthouse-core/scripts/chromium-web-tests/test.sh --reset-results
+```
+
+## How it works
```

Comment:
not a blocker, but how much of this can we upstream after this lands? Some of this (like the patch) sounds like it would be rough to fix if there's much drift in the DevTools repo and/or you're out for the week it breaks :)

Reply:
> Some of this (like the patch) sounds like it would be rough to fix if there's much drift in the DevTools repo and/or you're out for the week it breaks :)

Did a quick verification of this by changing the commit hash used to download blink_tools and seeing if it still worked. Not a bisect, just picked random commits from history.

```
7db009230db3d44dc184b4b38658b493f5e170aa Jul 28     (ToT, works)
41f18f569f022d78337f586dacc6eff0e0477e46 Jul 22     (currently used here, works)
ad976543abf3f4bccf88351a9ea93cd54511ad70 2019-12-18 (failed at first, but commenting out the same code in the patch was easy. code just moved around so patch was unable to apply)
f15fff781a560f28d82e8521fbfacecb4af1a567 2018-12-04 (failed with new errors; invalid VirtualTestSuites format)
```

It seems that unless the entire framework changes, the only issue will be the patch gets too stale to be applied.

> not a blocker, but how much of this can we upstream after this lands?

I wonder if a change such as "optionally require image_diff" or "optionally start apache" would even be entertained. It has no use within Chromium.

Here's my suggestion: we try to use the ToT blink_tools + this patch initially, and when that first breaks we hardcode a revision we know works while I attempt a new workaround. 
____

What I had in mind for some workarounds ...

We could get rid of the patches for `base.py` by stubbing the programs:

```
touch .tmp/chromium-web-tests/content-shells/791019/out/Release/image_diff

echo "#!/usr/bin/env bash\n\necho ok" > .tmp/chromium-web-tests/blink_tools/third_party/apache-mac/bin/httpd
chmod +x .tmp/chromium-web-tests/blink_tools/third_party/apache-mac/bin/httpd
```

Which just leaves `manager.py` calling `self._port.start_http_server`. `Port` has many different instances and is chosen in a factory class. Perhaps there's a way to provide our own Port instance, which stubs out this method?